### PR TITLE
SEQNG-228: Add the title to the seqexec engine

### PR DESCRIPTION
**Pull request by cquiroz**
This PR reads the sequence title when loading a sequence and can send it to the client

### CODE REVIEW

File: app/seqexec-server-gs/src/main/resources/app.conf [hunk 1]
```diff
@@ -30,6 +30,6 @@ seqexec-engine {
     f2Keywords = true
     gwskeywords = true
     odbQueuePollingInterval = "3 seconds"
-    tops = "tcs=tcs:, ao=ao:, gm=gm:, gc=gc:, gws= ws:, m2=m2:, oiwfs=oiwfs:, ag=ag:, f2=f2:"
+    tops = "tcs=tcs:, ao=ao:, gm=gm:, gc=gc:, gws=wos:, m2=m2:, oiwfs=oiwfs:, ag=ag:, f2=f2:"
```
Comment (jluhrs) [inline app/seqexec-server-gs/src/main/resources/app.conf]: gm=wos: ??

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala [hunk 2]
```diff
@@ -29,6 +34,8 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
     if (settings.odbNotifications) ODBProxy.OdbCommandsImpl(new Peer(settings.odbHost, 8442, null))
     else ODBProxy.DummyOdbCommands)
 
+  val odbClient = ODBClient(ODBClientConfig(settings.odbHost, ODBClient.DefaultODBBrowserPort))
```
Comment (jluhrs) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala]: Class OdbProxy exist as a façade for all interactions with the ODB. This odbClient should go there.

File: app/seqexec-server-gs/src/main/resources/app.conf (hunk 1 above)
Comment (cquiroz) [inline app/seqexec-server-gs/src/main/resources/app.conf]: Thanks, I only meant to remove the space

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala (hunk 2 above)
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala]: 👍

**Comment by cquiroz** [comment]
Fixed according to PR comments

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala [hunk 3]
```diff
@@ -122,10 +128,13 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
...
-      odbSeq <- Task(odbProxy.read(seqId))
-    } yield odbSeq.map(s => translator.sequence(translatorSettings)(seqId, s))
-    )
+    val t: EitherT[Task, SeqexecFailure, (List[SeqexecFailure], Option[Sequence[Action]])] = for {
+      odbSeq       <- EitherT(Task(odbProxy.read(seqId)))
+      progIdString <- EitherT(Task.delay(odbSeq.extract(OCS_KEY / InstConstants.PROGRAMID_PROP).as[String].leftMap(ConfigUtilOps.explainExtractError)))
+      progId       <- EitherT.fromTryCatchNonFatal(Task.now(SPProgramID.toProgramID(progIdString))).leftMap(e => SeqexecFailure.SeqexecException(e): SeqexecFailure)
+      name         <- EitherT(odbClient.observationTitle(progId, seqId.toString).map(_.leftMap(ConfigUtilOps.explainExtractError)))
+    } yield translator.sequence(translatorSettings)(seqId, odbSeq, name)
+
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala]: That sounds good. I just preserved the `EitherT` on the existing code

**Pull request merged**